snapshots/lcow/lcow.go

```diff
@@ -329,37 +329,48 @@ func (s *snapshotter) createSnapshot(ctx context.Context, kind snapshots.Kind, k
 		for _, o := range opts {
 			o(&snapshotInfo)
 		}
+		// IO/disk space optimization
+		//
+		// We only need one sandbox.vhd for the container. Skip making one for this
+		// snapshot if this isn't the snapshot that just houses the final sandbox.vhd
+		// that will be mounted as the containers scratch. Currently the key for a snapshot
+		// where a layer.vhd will be extracted to it will have the string `extract-` in it.
+		// If this is changed this will also need to be changed.
+		//
+		// We save about 17MB per layer (if the default scratch vhd size of 20GB is used) and of
+		// course the time to copy the vhd per snapshot.
+		if !strings.Contains(key, "extract-") {
```

Comment:
@dmcgowan - Can we get this to be a define somewhere to make this less fragile?

Reply:
Talked with @dcantah about adding a snapshot label when the active snapshot is just for extraction.

Reply:
Correct, I didn't get around to changing this PR yesterday but another approach to this is we could have a `WithScratchSnapshot` (or whatever we want to call it) snapshotter option we can pass in for the Prepare call for the r/w sandbox.vhdx snapshot that would just set a label we can check in the snapshotter to determine whether to make the scratch instead. This seems a little better than relying on the "extract-" key format staying static

Reply:
Ok after implementing it, I don't like the label approach either... Here's a branch with the work https://github.com/dcantah/containerd/commit/d570e8911bdb1530cdf0c1c979709803a44e7912 but let's see what @dmcgowan says about this approach first.

Reply:
@dmcgowan / @crosbymichael - Do either of you have an opinion on this or should we just stick with the key approach here? Thanks for the insights!

Reply:
I don't have an opinion on it.  I think we have done keys like this is the past so seems ok to me

Reply:
Thanks. SGTM too